testsuite/integration/src/test/java/org/keycloak/testsuite/broker/SAMLKeyCloakServerBrokerUserAttributeTest.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.broker;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.keycloak.common.util.Time;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.IdentityProviderRepresentation;
+import org.keycloak.services.managers.RealmManager;
+import org.keycloak.testsuite.KeycloakServer;
+import org.keycloak.testsuite.rule.AbstractKeycloakRule;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+
+public class SAMLKeyCloakServerBrokerUserAttributeTest extends AbstractIdentityProviderTest {
+
+    @ClassRule
+    public static AbstractKeycloakRule samlServerRule = new AbstractKeycloakRule() {
+
+        @Override
+        protected void configureServer(KeycloakServer server) {
+            server.getConfig().setPort(8082);
+        }
+
+        @Override
+        protected void configure(KeycloakSession session, RealmManager manager, RealmModel adminRealm) {
+            server.importRealm(getClass().getResourceAsStream("/broker-test/test-broker-realm-with-saml-with-multi-value-user-attributes.json"));
+        }
+
+        @Override
+        protected String[] getTestRealms() {
+            return new String[]{"realm-with-saml-idp-basic"};
+        }
+    };
+
+    @Override
+    protected String getProviderId() {
+        return "kc-saml-idp-basic";
+    }
+
+
+    @Test
+    public void canAccessMultiValueAttribute() {
+        UserModel user = assertSuccessfulAuthentication(getIdentityProviderModel(), "test-user-1", "new@email.com", true);
+        assertThat(user.getAttribute("mobile"), containsInAnyOrder("617-666-7777", "617-666-7778"));
+        assertThat(user.getFirstAttribute("mobile"), is("617-666-7777"));
+    }
+
+    @Test
+    public void canAccessSingleValueAttribute() {
+        UserModel user = assertSuccessfulAuthentication(getIdentityProviderModel(), "test-user-2", "new@email.com", true);
+        assertThat(user.getAttribute("mobile"), containsInAnyOrder("617-666-7777"));
+        assertThat(user.getFirstAttribute("mobile"), is("617-666-7777"));
+    }
+
+    @Test
+    public void nonExistingAttributeLookupReturnsAnEmptyList() {
+        UserModel user = assertSuccessfulAuthentication(getIdentityProviderModel(), "test-user-3", "new@email.com", true);
+        assertThat(user.getAttribute("mobile"), is(empty()));
+    }
+
+    @Test  //TODO
+    public void canUpdateBrokeredUserAttributes() {
```

Comment:
@hmlnarik I copied the code from AbstractKeycloakIdentityProviderTest but I am finding it a bit cryptic. Does it need to be this complex just to test that the attributes are updated through the mapper? Furthermore, I could not manage to assert that the attributes are there at the end of the test. Any suggestions? Thx